Author: Bernhard M. Wiedemann <bwiedemann suse de>
Date: 2017-06-29

make package build reproducible
by using fixed date + hostname

https://github.com/open-mpi/ompi/issues/3759
https://bugzilla.opensuse.org/show_bug.cgi?id=1047218 packages do not build reproducibly from including build time
https://bugzilla.opensuse.org/show_bug.cgi?id=1084909 +hostname

Index: openmpi-3.1.1.0.155d2134a776/autogen.pl
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/autogen.pl
+++ openmpi-3.1.1.0.155d2134a776/autogen.pl
@@ -85,7 +85,7 @@ if ($^O eq "solaris") {
 }
 
 $username = getpwuid($>);
-$full_hostname = `hostname`;
+$full_hostname = "openSUSEnohostname";
 chomp($full_hostname);
 $hostname = $full_hostname;
 $hostname =~ s/^([\w\-]+)\..+/\1/;
@@ -1190,7 +1190,7 @@ if (-e "orcm") {
 
 #---------------------------------------------------------------------------
 
-$full_hostname = `hostname`;
+$full_hostname = "openSUSEnohostname";
 chomp($full_hostname);
 
 $m4 = "dnl
Index: openmpi-3.1.1.0.155d2134a776/config/opal_functions.m4
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/config/opal_functions.m4
+++ openmpi-3.1.1.0.155d2134a776/config/opal_functions.m4
@@ -94,9 +94,9 @@ EOF
 # Save some stats about this build
 #
 
-OPAL_CONFIGURE_USER="`whoami`"
-OPAL_CONFIGURE_HOST="`(hostname || uname -n) 2> /dev/null | sed 1q`"
-OPAL_CONFIGURE_DATE="`date`"
+OPAL_CONFIGURE_USER="openSUSEnowhoami"
+OPAL_CONFIGURE_HOST="openSUSEnohostname"
+OPAL_CONFIGURE_DATE="openSUSEnodate"
 
 OPAL_LIBNL_SANITY_INIT
 
@@ -116,9 +116,9 @@ AC_DEFUN([OPAL_BASIC_SETUP],[
 # Save some stats about this build
 #
 
-OPAL_CONFIGURE_USER="`whoami`"
-OPAL_CONFIGURE_HOST="`(hostname || uname -n) 2> /dev/null | sed 1q`"
-OPAL_CONFIGURE_DATE="`date`"
+OPAL_CONFIGURE_USER="openSUSEnowhoami"
+OPAL_CONFIGURE_HOST="openSUSEnohostname"
+OPAL_CONFIGURE_DATE="openSUSEnodate"
 
 #
 # Make automake clean emacs ~ files for "make clean"
Index: openmpi-3.1.1.0.155d2134a776/config/opal_get_version.m4
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/config/opal_get_version.m4
+++ openmpi-3.1.1.0.155d2134a776/config/opal_get_version.m4
@@ -91,7 +91,7 @@ m4_define([OPAL_GET_VERSION],[
                     $2_REPO_REV=`git describe --tags --always`
                 fi
             else
-                $2_REPO_REV="date`date '+%Y-%m-%d'`"
+                $2_REPO_REV="date"
             fi
         fi
 
Index: openmpi-3.1.1.0.155d2134a776/ompi/tools/ompi_info/Makefile.am
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/ompi/tools/ompi_info/Makefile.am
+++ openmpi-3.1.1.0.155d2134a776/ompi/tools/ompi_info/Makefile.am
@@ -27,9 +27,9 @@ AM_CFLAGS = \
             -DOPAL_CONFIGURE_USER="\"@OPAL_CONFIGURE_USER@\"" \
             -DOPAL_CONFIGURE_HOST="\"@OPAL_CONFIGURE_HOST@\"" \
             -DOPAL_CONFIGURE_DATE="\"@OPAL_CONFIGURE_DATE@\"" \
-            -DOMPI_BUILD_USER="\"$$USER\"" \
-            -DOMPI_BUILD_HOST="\"`(hostname || uname -n) 2> /dev/null | sed 1q`\"" \
-            -DOMPI_BUILD_DATE="\"`date`\"" \
+            -DOMPI_BUILD_USER="\"USER\"" \
+            -DOMPI_BUILD_HOST="\"openSUSEnohostname\"" \
+            -DOMPI_BUILD_DATE="\"`date -u -r ../../../NEWS`\"" \
             -DOMPI_BUILD_CFLAGS="\"@CFLAGS@\"" \
             -DOMPI_BUILD_CPPFLAGS="\"@CPPFLAGS@\"" \
             -DOMPI_BUILD_CXXFLAGS="\"@CXXFLAGS@\"" \
Index: openmpi-3.1.1.0.155d2134a776/orte/tools/orte-info/Makefile.am
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/orte/tools/orte-info/Makefile.am
+++ openmpi-3.1.1.0.155d2134a776/orte/tools/orte-info/Makefile.am
@@ -24,9 +24,9 @@ AM_CFLAGS = \
             -DOPAL_CONFIGURE_USER="\"@OPAL_CONFIGURE_USER@\"" \
             -DOPAL_CONFIGURE_HOST="\"@OPAL_CONFIGURE_HOST@\"" \
             -DOPAL_CONFIGURE_DATE="\"@OPAL_CONFIGURE_DATE@\"" \
-            -DOMPI_BUILD_USER="\"$$USER\"" \
-            -DOMPI_BUILD_HOST="\"`(hostname || uname -n) | sed 1q`\"" \
-            -DOMPI_BUILD_DATE="\"`date`\"" \
+            -DOMPI_BUILD_USER="\"USER\"" \
+            -DOMPI_BUILD_HOST="\"openSUSEnohostname\"" \
+            -DOMPI_BUILD_DATE="\"`date -u -r ../../../NEWS`\"" \
             -DOMPI_BUILD_CFLAGS="\"@CFLAGS@\"" \
             -DOMPI_BUILD_CPPFLAGS="\"@CPPFLAGS@\"" \
             -DOMPI_BUILD_CXXFLAGS="\"@CXXFLAGS@\"" \
Index: openmpi-3.1.1.0.155d2134a776/oshmem/tools/oshmem_info/Makefile.am
===================================================================
--- openmpi-3.1.1.0.155d2134a776.orig/oshmem/tools/oshmem_info/Makefile.am
+++ openmpi-3.1.1.0.155d2134a776/oshmem/tools/oshmem_info/Makefile.am
@@ -16,9 +16,9 @@ AM_CPPFLAGS = \
             -DOPAL_CONFIGURE_USER="\"@OPAL_CONFIGURE_USER@\"" \
             -DOPAL_CONFIGURE_HOST="\"@OPAL_CONFIGURE_HOST@\"" \
             -DOPAL_CONFIGURE_DATE="\"@OPAL_CONFIGURE_DATE@\"" \
-            -DOMPI_BUILD_USER="\"$$USER\"" \
-            -DOMPI_BUILD_HOST="\"`(hostname || uname -n) 2> /dev/null | sed 1q`\"" \
-            -DOMPI_BUILD_DATE="\"`date`\"" \
+            -DOMPI_BUILD_USER="\"USER\"" \
+            -DOMPI_BUILD_HOST="\"openSUSEnohostname\"" \
+            -DOMPI_BUILD_DATE="\"`date -u -r ../../../NEWS`\"" \
             -DOMPI_BUILD_CFLAGS="\"@CFLAGS@\"" \
             -DOMPI_BUILD_CPPFLAGS="\"@CPPFLAGS@\"" \
             -DOMPI_BUILD_CXXFLAGS="\"@CXXFLAGS@\"" \
